yaks/lib/yaks/behaviour/optional_includes.rb

```diff
@@ -0,0 +1,31 @@
+require "rack/utils"
+
+module Yaks
+  module Behaviour
+    module OptionalIncludes
+      def self.included(base)
+        base.extend ClassMethods
+      end
+
+      module ClassMethods
+        def has_one(name, options = {}) # rubocop:disable Style/PredicateName
+          super name, options.merge(if: ->{include_association?(name)})
+        end
+
+        def has_many(name, options = {}) # rubocop:disable Style/PredicateName
+          super name, options.merge(if: ->{include_association?(name)})
+        end
```

Comment:
So what do we think about merging only if the key doesn't already exist in the options. That way we could always or never include a particular sub-resource simply by putting something in the config for `:if`

Can you see any issues with that approach?